syntax = "proto3";

package endpoints;

import "models/registered_function.proto";
import "models/value.proto";

message ExecuteFunctionRequest {
  // Required field specifying which function should be executed.
  optional models.RegisteredFunction function_to_execute = 1;

  // Required field representing the function parameter.
  optional models.Value parameter = 2;
}

message ExecuteFunctionResponse {
  // Required field representing the executed function result.
  optional models.Value result = 1;
}
